### PR TITLE
Fix regression in AnsiCastOpSuite with Spark 3.3.0

### PR DESCRIPTION
**Pull request by andygrove**
Closes https://github.com/NVIDIA/spark-rapids/issues/5457

AnsiCastOpSuite has assertions that look for an ANSI cast in a query plan. Prior to Spark 3.3.0 we could use toString to see if the name of the cast was "cast" or "ansi_cast" but now the name is always "cast" so we need to use reflection to access the protected field "ansiEnabled".

### CODE REVIEW

File: tests/src/test/scala/com/nvidia/spark/rapids/AnsiCastOpSuite.scala [hunk 1]
```diff
@@ -795,11 +795,20 @@ class AnsiCastOpSuite extends GpuExpressionTestSuite {
...
+      m.setAccessible(true)
+      m.getBoolean(c)
+    }
+
     val count = projections.map {
         case p: ProjectExec => p.projectList.count {
-          // ansiEnabled is protected so we rely on CastBase.toString
-          case c: CastBase => c.toString().startsWith("ansi_cast")
-          case Alias(c: CastBase, _) => c.toString().startsWith("ansi_cast")
+          case c: CastBase => isAnsiCast(c)
```
[the remark this answers is not in the document]
Comment (andygrove) [reply tests/src/test/scala/com/nvidia/spark/rapids/AnsiCastOpSuite.scala:810]: I will try that out. It is possible that the classes have changed over time so may require shimming but I am not 100% sure without looking into this more.

[the remark this answers is not in the document]
Comment (tgravescs) [reply tests/src/test/scala/com/nvidia/spark/rapids/AnsiCastOpSuite.scala:810]: ok, we can just leave  this way then.

**Review by tgravescs — APPROVED** [review]

**Comment by andygrove** [comment]
build

**Comment by firestarman** [comment]
Merged to unblock 330 build, and 340 development.

**Pull request merged**